workspaces/local-cli/src/shared/coverage.ts

```diff
@@ -0,0 +1,389 @@
+
+    table.push([
+      `${endpointName}\n\n${coverage}`,
+      `${requests}\n\n${responses}`,
+    ]);
+  });
+
+  //print it out
+  cli.action.stop();
+
+  console.log(colors.bold(`\n\n API Coverage Report:`));
+
+  console.log(table.toString());
+
+  console.log('\n\n');
+
+  const totals = report.coverageCounts.getCount(
+    CoverageConcerns.TotalInteractions()
+  );
+
+  console.log(
+    colors.grey(`Based on ${totals} samples from task: ${colors.bold('start')}`)
+  );
+
+  let resultTable = new Table({
+    head: [
+      colors.grey.underline(''),
+      colors.grey.underline('Observed'),
+      colors.grey.underline('Expected'),
+      colors.grey.underline('Percent Coverage'),
+    ],
+  });
+
+  function colorProperly(full: boolean, partial: boolean, none: boolean) {
+    return (string: string): string => {
+      if (full) {
+        return string;
+      } else if (none) {
+        return makeRed(string);
+      } else if (partial) {
+        return makeYellow(string);
+      }
+      return string;
+    };
+  }
+
+  const bodyCoverageColorer = colorProperly(
+    totalN === totalD,
+    totalN < totalD,
+    totalN === 0
+  );
+
+  const endpointCoverageColorer = colorProperly(
+    totalEndpointN === totalEndpointD,
+    totalEndpointN < totalEndpointD,
+    totalEndpointN === 0
+  );
+
+  resultTable.push(
+    [
+      'Documented Body Coverage',
+      totalN.toString(),
+      totalD.toString(),
+      Math.round((totalN / (totalD || 1)) * 100).toString() + '%',
+    ].map(bodyCoverageColorer),
+    [
+      'Documented Endpoint Coverage',
+      totalEndpointN.toString(),
+      totalEndpointD.toString(),
+      Math.round((totalEndpointN / (totalEndpointD || 1)) * 100).toString() +
+        '%',
+    ].map(endpointCoverageColorer)
+  );
+
+  console.log(resultTable.toString());
+}
+
+interface KnownEndpoint {
+  method: string;
+  pathId: string;
+  fullPath: string;
+  descriptor: SpecEndpointDescriptor;
+}
+
+async function getSpecEndpoints(paths: IPathMapping): Promise<KnownEndpoint[]> {
+  const events = await fs.readJSON(paths.specStorePath);
+  const { eventStore, rfcState, rfcService, rfcId } = universeFromEvents(
+    events
+  );
+
+  const queries = Queries(eventStore, rfcService, rfcId);
+  const cachedQueryResults = getCachedQueryResults(queries);
+  const shapesResolvers = queries.shapesResolvers();
+  const endpoints = queries.endpoints();
+  const allEndpoints = endpoints.map((i: any) => ({
+    ...i,
+    fullPath: queries.absolutePath(i.pathId),
+    descriptor: createEndpointDescriptor(i, cachedQueryResults),
+  }));
+
+  return sortBy(allEndpoints, (i) => i.fullPath);
+}
+
+async function getReport(paths: IPathMapping, captureId: string): Promise<any> {
+  const capturesDirectory = path.join(paths.capturesPath, captureId);
+
+  const entries = await fs.readdir(capturesDirectory);
+
+  const coverageFiles = entries
+    .filter((x) => x.startsWith('coverage-'))
```

Comment:
this prefix should be extracted into a variable so it is linked to the other usages